### PR TITLE
Tests for blockquotes plugin

### PR DESCRIPTION
**Pull request by noelherrick**
This makes blockquotes behave a little better, and adds tests! It builds on #46.

Blockquotes should:
- Become paragraphs when backspace is pressed and the cursor is at the beginning of the first line
- Merge lines when backspace is pressed and the cursor is at the beginning of any other line besides the first
- Stay blockquotes when all the of the text is selected and backspace is pressed
- Add additional lines when enter is pressed. If there is text to the right of the cursor, that text should start the next line
- Become paragraphs when enter is pressed and the cursor is at the end of the last line

### CODE REVIEW

File: packages/slate-blockquotes/__tests__/index.js [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+      <paragraph>Some value</paragraph>
+      <blockquote>
+        <blockquote_line><cursor />A</blockquote_line>
+      </blockquote>
+    `);
+
+    editor.setValue(input);
+
+    editor.run("onKeyDown", createEvent.keyDown({ key: "backspace" }));
+    editor.deleteBackward();
```
Comment (noelherrick) [inline packages/slate-blockquotes/__tests__/index.js]: @brendancarney I couldn't think of another way to simulate a backspace being pressed that both sends the event to the plugin and deletes the text. Thoughts?

**Comment by noelherrick** [comment]
@brendancarney I assume we should stick with Yarn vs. NPM? I should always use NPM and gitignore yarn.lock?

File: packages/slate-blockquotes/src/create-commands.js [hunk 2]
```diff
@@ -1,24 +1,50 @@
...
       editor.unwrapBlockByKey(parent.key, { type: blocks.blockquote });
-      editor.setNodeByKey(block.key, { type: blocks.default });
+      parent.nodes.forEach(block => {
+        editor.setNodeByKey(block.key, { type: blocks.default });
+      });
     });
   }
 };
 
+// const unwrapBlockquote = ({ blocks }, editor, options = {}) => {
```
Comment (brendancarney) [inline packages/slate-blockquotes/src/create-commands.js]: Commented out code

File: packages/slate-blockquotes/__tests__/index.js (hunk 1 above)
Comment (brendancarney) [reply packages/slate-blockquotes/__tests__/index.js]: At the moment, I don't know the best way to do this. It's probably adding functionality to `createEvent`, but I'm not sure it's worth it at the moment.

**Comment by brendancarney** [comment]
Use `yarn`, keep `yarn.lock`

**Comment by ishannonqa** [comment]
OK, I exercised this (https://deploy-preview-49--convertkit-slate-plugins.netlify.com/) with as many variations as I could think of, including the items noted in the summary.
Everything looks good.
QA Complete.

**Pull request merged**